packages/completer-extension/src/index.ts

```diff
@@ -45,337 +41,175 @@ namespace CommandIDs {
   export const selectFile = 'completer:select-file';
 }
 
-/**
- * A plugin providing code completion for editors.
- */
-const manager: JupyterFrontEndPlugin<ICompletionManager> = {
-  id: '@jupyterlab/completer-extension:manager',
-  autoStart: true,
-  provides: ICompletionManager,
-  activate: (app: JupyterFrontEnd): ICompletionManager => {
-    const handlers: { [id: string]: CompletionHandler } = {};
-
-    app.commands.addCommand(CommandIDs.invoke, {
-      execute: args => {
-        const id = args && (args['id'] as string);
-        if (!id) {
-          return;
-        }
-
-        const handler = handlers[id];
-        if (handler) {
-          handler.invoke();
-        }
-      }
-    });
-
-    app.commands.addCommand(CommandIDs.select, {
-      execute: args => {
-        const id = args && (args['id'] as string);
-        if (!id) {
-          return;
-        }
-
-        const handler = handlers[id];
-        if (handler) {
-          handler.completer.selectActive();
-        }
-      }
-    });
-
-    return {
-      register: (
-        completable: ICompletionManager.ICompletable,
-        renderer: Completer.IRenderer = Completer.defaultRenderer
-      ): ICompletionManager.ICompletableAttributes => {
-        const { connector, editor, parent } = completable;
-        const model = new CompleterModel();
-        const completer = new Completer({ editor, model, renderer });
-        const handler = new CompletionHandler({
-          completer,
-          connector
-        });
-        const id = parent.id;
-
-        // Hide the widget when it first loads.
-        completer.hide();
-
-        // Associate the handler with the parent widget.
-        handlers[id] = handler;
-
-        // Set the handler's editor.
-        handler.editor = editor;
-
-        // Attach the completer widget.
-        Widget.attach(completer, document.body);
-
-        // Listen for parent disposal.
-        parent.disposed.connect(() => {
-          delete handlers[id];
-          model.dispose();
-          completer.dispose();
-          handler.dispose();
-        });
-
-        return handler;
-      }
-    };
-  }
-};
+const COMPLETION_MANAGER_PLUGIN = '@jupyterlab/completer-extension:tracker';
 
-/**
- * An extension that registers consoles for code completion.
- */
-const consoles: JupyterFrontEndPlugin<void> = {
-  id: '@jupyterlab/completer-extension:consoles',
-  requires: [ICompletionManager, IConsoleTracker],
+const defaultProvider: JupyterFrontEndPlugin<void> = {
+  id: '@jupyterlab/completer-extension:base-service',
+  requires: [ICompletionProviderManager],
   autoStart: true,
   activate: (
     app: JupyterFrontEnd,
-    manager: ICompletionManager,
-    consoles: IConsoleTracker
+    serviceManager: ICompletionProviderManager
```

Comment:
Maybe `serviceManager` can be renamed to `completionManager` for clarity?

Since `serviceManager` usually refers to the `ServiceManager.IManager`.